### DISCUSSION TITLE
Using custom "Devices values configuration"s

### DISCUSSION THREAD

**Discussion by goodnewz** [zwave-js/zwave-js-ui#3613] [Q&A] [answered]
Hi there, 

I have a few devices on my Zwave network that I would like to all have the same configuration parameters. I saw in the settings menu that there is an option called "Devices values configuration," which appears to do what I want. However, it is not clear when or how these settings are applied, and the documentation is not clear. Does it happen only when adding a new device or at some interval, or maybe there is a need for a manual way to trigger this "update"? So far, just saving the settings has not changed anything. 

I would love someone with more experience with that setting to weigh in. I have attached a screenshot of the functionality I refer to in my text. 

![image](https://github.com/zwave-js/zwave-js-ui/assets/25808527/ccae503e-78a6-4015-a76b-c6b7c032fe83)

**Comment by kpine** [comment, marked as answer]
> I have a few devices on my Zwave network that I would like to all have the same configuration parameters. I saw in the settings menu that there is an option called "Devices values configuration," which appears to do what I want.

"Devices values configuration" does not have any specific relation to Z-Wave configuration parameters (documentation is at https://zwave-js.github.io/zwave-js-ui/#/usage/setup?id=device-values-configuration). "Configuration" here refers to the "configuration" of "device values", which are all of the things you see in the control panel. The fact that any value is available, not just configuration values, is a reflection of that.

It basically applies three different kinds of operations on "values" (copied from docs):

* Parse send: Enable this to allow users to specify a custom function(value,valueId,node,logger) to parse the value sent to MQTT. The function must be sync
* Parse receive: Enable this to allow users to specify a custom function(value,valueId,node,logger) to parse the value received via MQTT. The function must be sync
* Enable Poll: Enable poll of this value by using Z-Wave JS [pollValue](https://zwave-js.github.io/node-zwave-js/#/api/node?id=pollvalue)

The first two operations are only for MQTT and affect the values either received from the topic (parse receive) or values published to the topic (parse send). The last operation just enables polling for the selected values (refresh value from Z-Wave network).

If you want to automate the setting of configuration parameters for your device, you'll need to do that yourself, either through automations in a home automation software that [integrates](https://www.home-assistant.io/integrations/zwave_js/) with ZUI, using the [MQTT API](https://zwave-js.github.io/zwave-js-ui/#/guide/mqtt) manually, or running [driver functions](https://zwave-js.github.io/zwave-js-ui/#/usage/driver_function?id=driver-function) from Z-Wave JS UI.

For example, there is a "clone-config" driver function snippet built-in to ZUI. You can configure one device to the desired state, and run the snippet to clone it to other devices.

![image](https://github.com/zwave-js/zwave-js-ui/assets/791794/49360825-e36e-4954-9510-fc445578b578)

![image](https://github.com/zwave-js/zwave-js-ui/assets/791794/24c95417-e25f-4ba0-9f2d-6fd428586a95)

**Reply by kpine** [reply]
Related issues:
* https://github.com/zwave-js/zwave-js-ui/issues/1648
* https://github.com/zwave-js/zwave-js-ui/issues/1918

**Reply by goodnewz** [reply]
I really appreciate your detailed response @kpine. I completely misunderstood the concepts described in the documentation. Your suggestion on using `clone-config` supper helpful. Thanks for setting me on the right track!
